### DISCUSSION TITLE
Set merge_language for Mailchimp/Mandrill

### DISCUSSION THREAD

**Discussion by dbuckd** [anymail/django-anymail#234] [Q&A] [answered]
I'm trying to set the merge_language to handlebars for our Mailchimp templates. In testing I've tried changing the templates to the mailchimp default merge_language of `*|field|*` but because of dependability with some other platforms we require the handlebar `{{field}}` format.

I've tried to set using esp_extra:
`msg.esp_extra = {'merge_language': 'handlebars'}`

Also by setting the header as defined in the Mailchimp docs:
`msg = EmailMultiAlternatives(subj, body, from, to, header={'X-MC-MergeLanguage': 'handlebars'}`

Also via extra_headers
`msg.extra_headers["X-MC-MergeLanguage"] = 'handlebars'`

I can see that these are being sent over by Mailchimps API logs, so my guess is that the problem is more with Mailchimp than anymail. But if anyone has any ideas or further input I would greatly appreciate it.

**Comment by medmunds** [comment, marked as answer]
> I've tried to set using esp_extra:
> `msg.esp_extra = {'merge_language': 'handlebars'}`

From the [Mandrill docs](https://mailchimp.com/developer/transactional/api/messages/send-new-message/), it looks like `merge_language` has to be set within the `message` key, not as a top-level API parameter. Try this:

```python
msg.esp_extra = {'message': {'merge_language': 'handlebars'}}
```

Anymail's info on [`esp_extra` for Mandrill](https://anymail.readthedocs.io/en/stable/esps/mandrill/#esp-extra-support) goes into a little more detail on the different levels of parameters in MailChimp's transactional API.

[As with all things Mandrill, I haven't tested this myself, due to MailChimp's policy against providing development/test accounts.]

**Reply by dbuckd** [reply]
I just modified the msg.esp_extra field as you stated and success!

Thank you very much!
